projects/core/src/cart/facade/cart.service.ts

```diff
@@ -223,4 +225,50 @@ export class CartService {
   private isLoggedIn(userId: string): boolean {
     return typeof userId !== 'undefined';
   }
+
+  // TODO: Remove once backend is updated
+  /**
+   * Temporary method to merge guest cart with user cart because of beackend limitation
+   * This is for an edge case
+   */
+  private guestCartMerge(): void {
+    this.getEntries()
+      .pipe(take(1))
+      .subscribe(entries => {
+        this.store.dispatch(
+          new CartActions.DeleteCart({
+            userId: ANONYMOUS_USERID,
+            cartId: this.cartData.cart.guid,
+          })
+        );
+
+        combineLatest([
```

Comment:
I think, when user go to "checkout" process, that means he is satisfied with his existing cart. So, how about, instead of loading the user's latest cart, we create a new cart for the user.
The steps are:
1, delete the existing guest cart
2, create a new cart for the login user
3, add entries into the new cart.